token_claim_airdrop_transaction_e2e_test.go

```diff
@@ -0,0 +1,544 @@
+	env := NewIntegrationTestEnv(t)
+	defer CloseIntegrationTestEnv(env, nil)
+
+	// Create fungible token
+	tokenID, err := createFungibleToken(&env)
+	require.NoError(t, err)
+
+	// Create receiver with 0 auto associations
+	receiver, _ := createAccountHelper(t, &env, 0)
+
+	// Airdrop the tokens
+	airdropTx, err := NewTokenAirdropTransaction().
+		AddTokenTransfer(tokenID, receiver, transferAmount).
+		AddTokenTransfer(tokenID, env.OperatorID, -transferAmount).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	record, err := airdropTx.SetValidateStatus(true).GetRecord(env.Client)
+	require.NoError(t, err)
+
+	// Claim the tokens with the operator which does not have pending airdrops
+	// Fails with INVALID_SIGNATURE
+	claimResp, err := NewTokenClaimAirdropTransaction().
+		AddPendingAirdropId(record.PendingAirdropRecords[0].GetPendingAirdropId()).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	_, err = claimResp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.ErrorContains(t, err, "INVALID_SIGNATURE")
+}
+
+func TestIntegrationTokenClaimAirdropCannotClaimAlreadyClaimedAirdrop(t *testing.T) {
+	env := NewIntegrationTestEnv(t)
+	defer CloseIntegrationTestEnv(env, nil)
+
+	// Create fungible token
+	tokenID, err := createFungibleToken(&env)
+	require.NoError(t, err)
+
+	// Create receiver with 0 auto associations
+	receiver, receiverKey := createAccountHelper(t, &env, 0)
+
+	// Airdrop the tokens
+	airdropTx, err := NewTokenAirdropTransaction().
+		AddTokenTransfer(tokenID, receiver, transferAmount).
+		AddTokenTransfer(tokenID, env.OperatorID, -transferAmount).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	record, err := airdropTx.SetValidateStatus(true).GetRecord(env.Client)
+	require.NoError(t, err)
+
+	// Claim the tokens with the receiver
+	claimTx, err := NewTokenClaimAirdropTransaction().
+		AddPendingAirdropId(record.PendingAirdropRecords[0].GetPendingAirdropId()).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+
+	claimResp, err := claimTx.Sign(receiverKey).Execute(env.Client)
+	require.NoError(t, err)
+
+	_, err = claimResp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// Claim the tokens with the receiver again
+	// Fails with INVALID_PENDING_AIRDROP_ID
+	claimTx, err = NewTokenClaimAirdropTransaction().
+		AddPendingAirdropId(record.PendingAirdropRecords[0].GetPendingAirdropId()).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+
+	claimResp, err = claimTx.Sign(receiverKey).Execute(env.Client)
+	require.NoError(t, err)
+
+	_, err = claimResp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.ErrorContains(t, err, "INVALID_PENDING_AIRDROP_ID")
+}
+
+func TestIntegrationTokenClaimAirdropCannotClaimWithEmptyPendingAirdrops(t *testing.T) {
+	env := NewIntegrationTestEnv(t)
+	defer CloseIntegrationTestEnv(env, nil)
+
+	// Claim the tokens with the receiver without setting pendingAirdropIds
+	// Fails with EMPTY_PENDING_AIRDROP_ID_LIST
```

Comment:
nit: I think this is obvious but you can leave it as is just marking. Would only lead to a risk of mismatch of the comment and the actual string type if modified by services.

Reply:
Removed